RadixWallet/Features/SettingsFeature/Preferences/HiddenAssets/HiddenAssets+View.swift

```diff
@@ -0,0 +1,139 @@
+import SwiftUI
+
+// MARK: - HiddenAssets.View
+extension HiddenAssets {
+	public struct View: SwiftUI.View {
+		public let store: StoreOf<HiddenAssets>
+
+		public var body: some SwiftUI.View {
+			WithPerceptionTracking {
+				ScrollView {
+					LazyVStack(alignment: .leading, spacing: .large3) {
+						Text(L10n.HiddenAssets.text)
+							.textStyle(.body1HighImportance)
+							.foregroundColor(.app.gray2)
+
+						header(L10n.HiddenAssets.fungibles)
+						fungibles
+
+						header(L10n.HiddenAssets.nonFungibles)
+						nonFungibles
+
+						header(L10n.HiddenAssets.poolUnits)
+						poolUnits
+					}
+					.padding(.medium3)
+				}
+				.background(Color.app.gray5)
+				.radixToolbar(title: L10n.HiddenAssets.title)
+				.task {
+					store.send(.view(.task))
+				}
+				.alert(store: store.scope(state: \.$destination.unhideAlert, action: \.destination.unhideAlert))
+			}
+		}
+
+		private func header(_ value: String) -> some SwiftUI.View {
+			Text(value)
+				.textStyle(.secondaryHeader)
+				.foregroundColor(.app.gray2)
+		}
+
+		@ViewBuilder
+		private var fungibles: some SwiftUI.View {
+			if store.fungible.isEmpty {
```

Comment:
annoying that ResultBuilders does not support guard or otherwise return early :/ 
[SE-0289 states](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0289-result-builders.md)

> `guard` is provisionally ill-formed when it appears within a transformed function. Situations in which this statement may appear may be supported in the future, such as when the statement does not produce a partial result.

Reply:
right, or ternary operators so that we could do something like this on the `body`:
```Swift
store.fungibles.isEmpty ? empty : fungibles
```

and remove the conditional check inside `var fungibles`

Reply:
You could put the if-else at the parent level, and let the fungibles var only be called when not empty. That would be pretty clean. I think it should be called `emptyState` though, not just `empty`. And I would make it a nested view, so you would use it with `EmptyState()` instead, it doesn't seem to use any variables.

Reply:
well this is debatable. I prefer looking at the body and getting a clear and quick vision of what's going on. In my opinion it looks cleaner to see this:
```Swift
LazyVStack(alignment: .leading, spacing: .large3) {
	Text(L10n.HiddenAssets.text)
		.textStyle(.body1HighImportance)
		.foregroundColor(.app.gray2)

	header(L10n.HiddenAssets.fungibles)
	fungibles

	header(L10n.HiddenAssets.nonFungibles)
	nonFungibles

	header(L10n.HiddenAssets.poolUnits)
	poolUnits
}
```

than this:
```Swift
LazyVStack(alignment: .leading, spacing: .large3) {
	Text(L10n.HiddenAssets.text)
		.textStyle(.body1HighImportance)
		.foregroundColor(.app.gray2)

	header(L10n.HiddenAssets.fungibles)
	if store.fungibles.isEmpty {
		empty
	} else {
		fungibles
	}

	header(L10n.HiddenAssets.nonFungibles)
	if store.nonFungibles.isEmpty {
		empty
	} else {
		nonFungibles
	}

	header(L10n.HiddenAssets.poolUnits)
	if store.poolUnits.isEmpty {
		empty
	} else {
		poolUnits
	}
}
```

Reply:
Sure, but the second option makes the variables cleaner. But my main point is that it is better to not hide the empty state, since that is in fact a top-level thing, I don't think it should be hidden.